Makefile

```diff
@@ -255,8 +255,21 @@ deploy-aws-account-operator:
 # Create zero size account pool
 	@oc apply -f hack/files/aws_v1alpha1_zero_size_accountpool.yaml
 
-.PHONY: deploy
-deploy: check-aws-credentials deploy-aws-account-operator-credentials deploy-aws-account-operator
+.PHONY: deploy-local
```

Comment:
Since the sub-targets are using `oc apply`, those steps are idempotent, so running them each time is a no-op -- *unless* you've changed your secrets or those resource defs, which is also a thing you might be doing while developing. So I'd definitely like to keep umbrella targets that do both, if possible.

If you really want to keep a way to skip the setup steps, I split it out again. Naming-wise, I feel like a target with `deploy` in the name should actually wind up with the operator deployed. So I could
- rename the existing `deploy` (`prepare`? `predeploy`? `your-suggestion-here`?)
- (optionally) create new targets (`up-local`/`up-cluster`? `run-local`/`run-cluster`? modern operator-sdk CLI replaced `up` with `run` so the latter might be more forward-looking) that *only* run the actual operator
- keep these `deploy-local` and `deploy-cluster` targets doing the same thing they're doing now, i.e. {default `FORCE_DEV_MODE`, predeploy, run}, by subsuming whatever targets we end up with above.

Thoughts?

Reply:
I guess the only real problem I'm going to have is that if the aws credentials are not stored in your environment then the deploy will fail, forcing you to keep the specific AWS credentials in your env instead of just a set-and-forget mentality because that's what the `check-aws-credentials` make target is doing.

However, with that said we could do something like modify the `check-aws-credentials` and the `deploy-aws-account-operator-credentials` to no-op if the secret is already present.  Or if we want to consider utilizing these to change the workflow it could be something like this:

* `check-aws-credentials` will exit cleanly if the aws credentials are present OR the secret is set.
* `deploy-aws-account-operator-credentials` will no-op if the secret is set AND the env vars are not set, otherwise it will replace the secret if the secret is present and the env vars are set, OR it will create the secret if it's not present and the env vars are set.

And then it will no-op instead of erroring for each run.

And honestly, even if we _don't_ do all of that work, at the end of the day you can just run `make deploy-local` and then continue with running `operator-sdk up local` the same way everyone's doing it now anyway.

Reply:
Okay, I think I made this happy without having a zillion targets doing slightly different things. In the latest commit:
- `make predeploy` does what `deploy` did before. It's the (only) thing that needs the creds env vars.
- `make deploy-local` and `make deploy-cluster` only do the bringing-up-the-actual-operator bits.
So the expected workflow is now:
```
make predeploy
make deploy-$whichever

while not done:
    ... develop ...
    make deploy-$whichever
```

Reply:
Nice.  I'm not going to hold this up anymore but I would personally prefer `make install` instead of `make pre-deploy`. I'll ultimately leave that decision up to you, but this works for me :)

Reply:
Thinking it through, I would like to keep the $verb the same for all the targets, to make it clear how they're related. So either `preinstall` `install-cluster` `install-local` or `predeploy` `deploy-local` `deploy-cluster`.

Unless there's some nuanced definition of "install" vs "deploy" that I'm missing?